ansible/monitoring.yml

```diff
@@ -0,0 +1,37 @@
+- name: Configure monitoring services (DataDog)
+  hosts: all
+  remote_user: master
+  become: yes
+  tags: [datadog]
+
+  vars:
+    datadog_config:
+      apm_enabled: false
+      use_dogstatsd: yes
+      process_agent_enabled: true
+      hostname: "{{ ansible_fqdn }}"
+      tags: "{{ group_names|join(',') }}"
+    datadog_checks:
+      system_core:
+        init_config:
+        instances:
+          # The Agent just needs one item in instances in order to enable the check.
+          # The content of the item doesn’t matter.
+          - foo: bar
+      process:
+        init_config:
+        instances:
+          - name: ssh
+            search_string: ['sshd']
+          - name: epoch
+            search_string: ['beam.smp']
```

Comment:
I'm not sure if this is the best process name to monitor

Reply:
This looks correct. Though I see two potential issues:
1. If operator needs to attach to node for inspecting state, I expect he/she will use e.g. `bin/epoch remote_console` that rely on `epmd` process not having died since the epoch node started;
2. If operator needs to attach to node for inspecting state e.g. `bin/epoch remote_console` this will start a second beam.smp process: does datadog get confused?

I am checking solutions for the two problems.

Reply:
> 2. If operator needs to attach to node for inspecting state e.g. bin/epoch remote_console this will start a second beam.smp process: does datadog get confused?

Below is an example of the two beam.smp processes as returned by `ps aux` - the best difference appears to be the `-boot /tmp/node/releases/0.3.3/start` part (after beam.smp):
```
ubuntu    7089  0.4  7.6 2695292 77656 pts/1   Ssl+ 10:58   0:07 /tmp/node/erts-9.0.1/bin/beam.smp -K true -A30 -SDio 10 -- -root /tmp/node -progname tmp/node/bin/epoch -- -home /home/ubuntu -- -boot /tmp/node/releases/0.3.3/start -mode embedded -boot_var ERTS_LIB_DIR /tmp/node/erts-9.0.1/../lib -config /tmp/node/releases/0.3.3/sys.config -sname epoch -setcookie epoch_cookie -- console
ubuntu    8176  0.1  2.7 2629512 28440 pts/0   Sl+  11:24   0:00 /tmp/node/erts-9.0.1/bin/beam.smp -- -root /tmp/node -progname erl -- -home /home/ubuntu -- -sname remsh42e4e36b-epoch@ubuntu-xenial -remsh epoch@ubuntu-xenial -boot start_clean -boot_var ERTS_LIB_DIR /tmp/node/erts-9.0.1/../lib -setcookie epoch_cookie -hidden -kernel net_ticktime 60
```

I did not find the reference of these `datadog_checks` so not sure how to express this in the config.

Reply:
> 1. If operator needs to attach to node for inspecting state, I expect he/she will use e.g. bin/epoch remote_console that rely on epmd process not having died since the epoch node started;

This looks like not a monitoring/datadog issue but rather something that shall be done in the init supervision tree. I suggest amending datadog config with monitoring if epmd process is up, and warning (not error) if that is not the case.

Reply:
I assume that more than one epmd process is not expected?

Also why warning instead of error ?

Reply:
> I assume that more than one epmd process is not expected?

Correct.

Reply:
> Also why warning instead of error? Missing epmd will render the node in "non-running" state.

I understand that if epmd died the node will keep running and working. Am I missing something? The issue will be for operator not being able to use `bin/epoch ping` or `bin/epoch remote_console`.

Reply:
I don't have much understanding how erlang handle it's processes, so I assume your comment is correct.

So, you basically asked me to add additional check for empd (warning) next to beam.smp ?

Reply:
see 183187f